test/UnitTest/Microsoft.AspNetCore.OData.Test/Abstraction/ODataActionSelectorTestHelper.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using System.IO;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using System.Linq;
+#if NETCOREAPP2_0
+using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.OData;
+    using Microsoft.AspNetCore.Mvc.Internal;
+#else
+using Moq;
+#endif
+
+
+namespace Microsoft.AspNet.OData.Test.Abstraction
+{
+    /// <summary>
+    /// Helper methods for working with ODataActionSelector tests
+    /// </summary>
+    public static class ODataActionSelectorTestHelper
+    {
+        /// <summary>
+        /// Checks whether the specified method is a suitable match
+        /// for the specified action
+        /// </summary>
+        /// <param name="action">Controller action to match</param>
+        /// <param name="method">Candidate method to compare</param>
+        /// <returns></returns>
+        public static bool ActionMatchesMethod(ActionDescriptor action, MethodInfo method)
+        {
+            if (action.DisplayName != method.Name)
+            {
+                return false;
+            }
+
+            var parameters = method.GetParameters();
+            if (parameters.Length != action.Parameters.Count)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < parameters.Length; i++)
+            {
+                if (parameters[i].Name != action.Parameters[i].Name ||
```

Comment:
@habbes I know this class is for test purposes but I hope that if in the actual logic the parameter types are playing any part we're careful with the comparison so we don't trip at implicit conversion... e.g. from `int` to `long`

Reply:
This is not used in the actual logic, it's just used to compare the selected controller action with the expected controller method for the test assertions.